native/core/src/parquet/objectstore/s3.rs

```diff
@@ -107,6 +111,37 @@ pub fn create_store(
     Ok((Box::new(object_store), path))
 }
 
+/// Get the bucket region using the [HeadBucket API]. This will fail if the bucket does not exist.
+///
+/// [HeadBucket API]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html
+pub async fn resolve_bucket_region(bucket: &str) -> Result<String, Box<dyn Error>> {
```

Comment:
Did we log an issue to improve the error reporting in the original implementation as well? If so can we add a comment here to track that?

Reply:
Yes, I have now added a link to https://github.com/apache/arrow-rs-object-store/issues/479